tensorrt_llm/_torch/attention_backend/fmha/flashinfer_trtllm_gen.py

```
@@ -67,6 +67,80 @@
+                self.input_idx == other.input_idx
+                and self.dim_idx == other.dim_idx
+                and buckets_equal
+                and self.map_to_tuning_buckets is other.map_to_tuning_buckets
+            )
+
+        DynamicTensorSpec.__eq__ = _hash_consistent_eq
+        logger.debug(
+            "Patched flashinfer DynamicTensorSpec.__eq__ to be hash-consistent "
+            "(autotuner cache collision-chain workaround)."
+        )
+    except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)